### ISSUE TITLE
mAP problem

### ISSUE THREAD

**Issue by Yachumika** [Tianxiaomo/pytorch-YOLOv4#225]
As  you can see  in the red region . Anyone had the same problem? I really need the help!!  

![S__24985612](https://user-images.githubusercontent.com/55393406/89972802-55701980-dc91-11ea-8df7-220aea9afd29.jpg)

**Comment by JiangMei0201** [comment]
> As you can see in the red region . Anyone had the same problem? I really need the help!!
> 
> ![S__24985612](https://user-images.githubusercontent.com/55393406/89972802-55701980-dc91-11ea-8df7-220aea9afd29.jpg)
Did you use coco dataset? Have you solved your problem？

**Comment by Yachumika** [comment]
> > As you can see in the red region . Anyone had the same problem? I really need the help!!
> > ![S__24985612](https://user-images.githubusercontent.com/55393406/89972802-55701980-dc91-11ea-8df7-220aea9afd29.jpg)
> > Did you use coco dataset? Have you solved your problem？

No, i havn't solve it . I use VOC dataset . I still don't understand why...

**Comment by JiangMei0201** [comment]
我没有遇见相同的问题。我的get_image_id函数就有问题，所以evaluate直接报错。你需要测试一下，权值训练的怎么样。我是训练结果很好，只是evaluate一直报错（所以之前我就注释掉了测试部分）




    


On 08/17/2020 13:28, YA JHU-YANG wrote: 


As you can see in the red region . Anyone had the same problem? I really need the help!!

Did you use coco dataset? Have you solved your problem？


No, i havn't solve it . I use VOC dataset . I still don't understand why...

—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/Tianxiaomo/pytorch-YOLOv4/issues/225#issuecomment-674663725",
"url": "https://github.com/Tianxiaomo/pytorch-YOLOv4/issues/225#issuecomment-674663725",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by Yachumika** [comment]
> 我没有遇见相同的问题。我的get_image_id函数就有问题，所以evaluate直接报错。你需要测试一下，权值训练的怎么样。我是训练结果很好，只是evaluate一直报错（所以之前我就注释掉了测试部分） On 08/17/2020 13:28, YA JHU-YANG wrote: As you can see in the red region . Anyone had the same problem? I really need the help!! Did you use coco dataset? Have you solved your problem？ No, i havn't solve it . I use VOC dataset . I still don't understand why... —You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe. [ { "@context": "http://schema.org", "@type": "EmailMessage", "potentialAction": { "@type": "ViewAction", "target": "[#225 (comment)](https://github.com/Tianxiaomo/pytorch-YOLOv4/issues/225#issuecomment-674663725)", "url": "[#225 (comment)](https://github.com/Tianxiaomo/pytorch-YOLOv4/issues/225#issuecomment-674663725)", "name": "View Issue" }, "description": "View this Issue on GitHub", "publisher": { "@type": "Organization", "name": "GitHub", "url": "https://github.com" }

**Comment by JiangMei0201** [comment]
> > > As you can see in the red region . Anyone had the same problem? I really need the help!!
> > > ![S__24985612](https://user-images.githubusercontent.com/55393406/89972802-55701980-dc91-11ea-8df7-220aea9afd29.jpg)
> > > Did you use coco dataset? Have you solved your problem？
> 
> No, i havn't solve it . I use VOC dataset . I still don't understand why...
我不太清楚

> > > As you can see in the red region . Anyone had the same problem? I really need the help!!
> > > ![S__24985612](https://user-images.githubusercontent.com/55393406/89972802-55701980-dc91-11ea-8df7-220aea9afd29.jpg)
> > > Did you use coco dataset? Have you solved your problem？
> 
> No, i havn't solve it . I use VOC dataset . I still don't understand why...

Maybe you should make sure that your dataset could train a good weights(you can make # between train,py  line 415-440). I use my owe voc datasets

**Comment by Yachumika** [comment]
@JiangMei0201  thank u very much，how to make sure my datasets train a good weights ?  I really like to discuss more issues with you , if you want too, that is my email :  yang.yajhu@gmail.com.

**Comment by Jinhui-Qin** [comment]
@JiangMei0201 I got the same issue. And I double check my datasets which are absolutely right. So how to train a good weights?

**Comment by zhouzhubin** [comment]
@Yachumika 请问你解决了吗?我也遇到这个问题了,我的数据在tensorflow版本的yolo3是可以训练出来的,但是在这个工程上,ap一直为0的

**Comment by anheidelonghu** [comment]
Maybe you can set Inference=False when initial model in train.py

**Comment by shaojiestar** [comment]
I also meet this problem,and I don't know hao to resolve it ......My train/val dataset format at below:
img_path   x1,y1, x2,y2,class_id x1,y1,x2,y2,class_id ...
img_path   x1,y1, x2,y2,class_id x1,y1,x2,y2,class_id ...
but the evaluate metrics is always 0
Can you give me some guides?

**Comment by mfang2020** [comment]
I experienced the same issue (AP=0). My current guess is that the input format as told in the author's file "img_path x1,y1, x2,y2,class_id x1,y1,x2,y2,class_id ..." is VOC-format, while the default code is written for COCO-format. I've been trying to rewrite the `convert_to_xywh()` in coco_eval.py.

**Comment by Qianshaowei** [comment]
> I experienced the same issue (AP=0). My current guess is that the input format as told in the author's file "img_path x1,y1, x2,y2,class_id x1,y1,x2,y2,class_id ..." is VOC-format, while the default code is written for COCO-format. I've been trying to rewrite the `convert_to_xywh()` in coco_eval.py.

I think so too. Have you solved it now? I need your help.

**Comment by missFuture** [comment]
> I experienced the same issue (AP=0). My current guess is that the input format as told in the author's file "img_path x1,y1, x2,y2,class_id x1,y1,x2,y2,class_id ..." is VOC-format, while the default code is written for COCO-format. I've been trying to rewrite the `convert_to_xywh()` in coco_eval.py.

Hi, Do you solve the AP = 0 issue after rewrite the `convert_tp_xywh()` in coco_eval.py?  let me know more about this. thank you

**Comment by YuyangSunshine** [comment]
I have the same problem, but I'm sure that the setting of the input data has changed to COCO-format. Are you solve this problem?
